### PR TITLE
12.0 mig report xlsx

### PR DESCRIPTION
**Pull request by RickHunterEc**
Migrate v 12.0. Adapt _executeReportAction in qwebactionmanager.js for generate reports with new Odoo ActionManager

### CODE REVIEW

**Comment by RickHunterEc** [comment]
@pedrobaeza , please, can you review if any problem with Travis. Impossible create a new database for erorr in instance.  Attach log:
Creating instance:
unbuffer /home/travis/OCB-12.0/odoo-bin -d openerp_template --log-level=info --stop-after-init --init web,base
2018-10-13 08:20:09,146 4678 INFO ? odoo: Odoo version 12.0 
2018-10-13 08:20:09,147 4678 INFO ? odoo: Using configuration file at /home/travis/.openerp_serverrc 
2018-10-13 08:20:09,147 4678 INFO ? odoo: addons paths: ['/home/travis/data_dir/addons/12.0', '/home/travis/build/OCA/reporting-engine', '/home/travis/OCB-12.0/addons', '/home/travis/OCB-12.0/odoo/addons'] 
2018-10-13 08:20:09,147 4678 INFO ? odoo: database: default@default:default 
2018-10-13 08:20:09,709 4678 INFO ? odoo.addons.base.models.ir_actions_report: Will use the Wkhtmltopdf binary at /home/travis/maintainer-quality-tools/travis/wkhtmltopdf 
2018-10-13 08:20:09,971 4678 INFO ? odoo.modules.loading: init db 
2018-10-13 08:20:10,464 4678 INFO openerp_template odoo.modules.loading: loading 1 modules... 
2018-10-13 08:20:10,622 4678 INFO openerp_template odoo.modules.registry: module base: creating or updating database tables 
2018-10-13 08:20:11,343 4678 ERROR openerp_template odoo.sql_db: bad query: b"\n                INSERT INTO ir_model_data (module, name, model, res_id, noupdate, date_init, date_update)\n                VALUES ('base', 'field_base____last_update', 'ir.model.fields', 3, false, now() at time zone 'UTC', now() at time zone 'UTC'), ('base', 'field_base__id', 'ir.model.fields', 1, false, now() at time zone 'UTC', now() at time zone 'UTC'), ('base', 'field_base__display_name', 'ir.model.fields', 2, false, now() at time zone 'UTC', now() at time zone 'UTC')\n                ON CONFLICT (module, name)\n                DO UPDATE SET date_update=(now() at time zone 'UTC') \n            "
ERROR: syntax error at or near "ON"
LINE 4:                 ON CONFLICT (module, name)
                        ^
 
2018-10-13 08:20:11,343 4678 ERROR openerp_template odoo.addons.base.models.ir_model: Failed to insert ir_model_data
('base', 'field_base____last_update', 'ir.model.fields', 3, False)
('base', 'field_base__id', 'ir.model.fields', 1, False)
('base', 'field_base__display_name', 'ir.model.fields', 2, False) 
2018-10-13 08:20:11,344 4678 ERROR openerp_template odoo.sql_db: bad query: b"UPDATE ir_module_module SET state='installed' WHERE state IN ('to remove', 'to upgrade')"
ERROR: relation "ir_module_module" does not exist
LINE 1: UPDATE ir_module_module SET state='installed' WHERE state IN...
               ^

**Comment by RickHunterEc** [comment]
Hi, @pedrobaeza , please, can you check the Travis error. Database creation failed when install base module, but in runbot the module is ok. The log is:
2018-10-13 08:20:11,343 4678 ERROR openerp_template odoo.sql_db: bad query: b"\n                INSERT INTO ir_model_data (module, name, model, res_id, noupdate, date_init, date_update)\n                VALUES ('base', 'field_base____last_update', 'ir.model.fields', 3, false, now() at time zone 'UTC', now() at time zone 'UTC'), ('base', 'field_base__id', 'ir.model.fields', 1, false, now() at time zone 'UTC', now() at time zone 'UTC'), ('base', 'field_base__display_name', 'ir.model.fields', 2, false, now() at time zone 'UTC', now() at time zone 'UTC')\n                ON CONFLICT (module, name)\n                DO UPDATE SET date_update=(now() at time zone 'UTC') \n            "
ERROR: syntax error at or near "ON"
LINE 4:                 ON CONFLICT (module, name)

Thank you.

File: report_xlsx/README.rst [hunk 1]
```diff
@@ -0,0 +1,90 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
```
Comment (elicoidal) [inline report_xlsx/README.rst]: png

**Comment by pedrobaeza** [comment]
There's already another PR for migrating this module in #248 by @ovnicraft. You should both coordinate. This ones seems to be more polite, although:

- Why do you remove translations?
- Why do you duplicate commits? Squash all migration commits in one or at least avoid duplicated commits.

About the error in Travis, there was an incorrect PG version definition. If you rebase over current 12.0 branch, you will get that correct.

File: report_xlsx/README.rst [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+        def generate_xlsx_report(self, workbook, data, partners):
+            for obj in partners:
+                report_name = obj.name
+                # One sheet by partner
+                sheet = workbook.add_worksheet(report_name[:31])
+                bold = workbook.add_format({'bold': True})
+                sheet.write(0, 0, obj.name, bold)
+
+To manipulate the ``workbook`` and ``sheet`` objects, refer to the
+`documentation <http://xlsxwriter.readthedocs.org/>`_ of ``xlsxwriter``.
```
Comment (elicoidal) [inline report_xlsx/README.rst]: suggestion:
```
`xlsxwriter documentation <http://xlsxwriter.readthedocs.org/>`_

File: report_xlsx/README.rst [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+        string="Print to XLSX"
+        report_type="xlsx"
+        name="module_name.report_name"
+        file="res_partner"
+        attachment_use="False"
+    />
+
+.. image:: https://odoo-community.org/website/image/ir.attachment/5784_f2813bd/datas
+   :alt: Try me on Runbot
+   :target: https://runbot.odoo-community.org/runbot/143/11.0
```
Comment (elicoidal) [inline report_xlsx/README.rst]: 12

File: report_xlsx/__manifest__.py [hunk 4]
```diff
@@ -0,0 +1,31 @@
+# Copyright 2015 ACSONE SA/NV (<http://acsone.eu>)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+{
+    'name': "Base report xlsx",
```
Comment (elicoidal) [inline report_xlsx/__manifest__.py]: XLSX (multiple). This is an acronym

File: report_xlsx/report/report_xlsx.py [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+    _logger.debug('Can not import xlsxwriter`.')
+
+
+class ReportXlsxAbstract(models.AbstractModel):
+    _name = 'report.report_xlsx.abstract'
+    _description = 'Generate XLSX Reports'
+
+    def _get_objs_for_report(self, docids, data):
+        """
+        Returns objects for xlx report.  From WebUI these
```
Comment (elicoidal) [inline report_xlsx/report/report_xlsx.py]: xlsx

File: report_xlsx/static/src/js/report/qwebactionmanager.js [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+                }
+            });
+            framework.unblockUI();
+            var closeAction = { type: 'ir.actions.act_window_close' };
+            return self.doAction(closeAction, _.pick(options, 'on_close'));
+        }
+        return self._super(action, options);
+    }
+});
+});
```
Comment (elicoidal) [inline report_xlsx/static/src/js/report/qwebactionmanager.js:48]: line

File: report_xlsx/tests/test_report.py [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo.tests import common
+import logging
+_logger = logging.getLogger(__name__)
+
+try:
+    from xlrd import open_workbook
+except ImportError:
+    _logger.debug('Can not import xlsxwriter`.')
```
Comment (elicoidal) [inline report_xlsx/tests/test_report.py]: xlrd

**Review by elicoidal — CHANGES_REQUESTED** [review]
some details
Thanks for the contribution!

File: report_xlsx/README.rst (hunk 1 above)
Comment (RickHunterEc) [reply report_xlsx/README.rst]: > There's already another PR for migrating this module in #248 by @ovnicraft. You should both coordinate. This ones seems to be more polite, although:
> 
> * Why do you remove translations?
> * Why do you duplicate commits? Squash all migration commits in one or at least avoid duplicated commits.
> 
> About the error in Travis, there was an incorrect PG version definition. If you rebase over current 12.0 branch, you will get that correct.
Thanks for review @pedrobaeza. Remove temporaly traslantions for test modules with new nomenclature of i18n (model_terms in some fields).  The next commit will have traslations back.
@elicoidal Thanks for you review. I change now.

File: report_xlsx/static/src/js/report/qwebactionmanager.js (hunk 6 above)
Comment (RickHunterEc) [reply report_xlsx/static/src/js/report/qwebactionmanager.js:48]: Hi @elicoidal , done.

Comment (RickHunterEc) [reply report_xlsx/static/src/js/report/qwebactionmanager.js:48]: Changes done. Travis and Runbot test passed. 
Thanks.

**Review by elicoidal — APPROVED** [review]
LGTM but please attend @pedrobaeza concern

**Comment by RickHunterEc** [comment]
The PR https://github.com/OCA/reporting-engine/pull/248 is complete and refactoring for v. 12.

**Pull request closed without merging**